querybuilder/rule.go

```diff
@@ -19,36 +19,42 @@ type Rule struct {
 }
 
 // Evaluate function checks whether the dataset matches with rule
-func (r *Rule) Evaluate(dataset map[string]interface{}) bool {
+func (r *Rule) Evaluate(dataset map[string]interface{}) (bool, error) {
+	var err error = nil
 	var wg sync.WaitGroup
 	var input, value interface{}
 
 	opr, ok := operator.GetOperator(r.Operator)
 	if !ok {
-		return false
+		return false, err
 	}
 
 	wg.Add(2)
 	go func() {
-		input = r.getInputValue(dataset)
+		input, err = r.getInputValue(dataset)
 		wg.Done()
 	}()
 
 	go func() {
-		value = r.getValue()
+		value, err = r.getValue()
```

Comment:
here we are doing synchronization using `sync.WaitGroup`. If you are using an `err` to store the erros, it will get overwritten by other `goroutines`. 

prefer to handle errors in goroutes during synchronization using `errgroup.Group` instead.